src/main/java/edu/cornell/kfs/tax/batch/service/impl/TaxPayeeHelperServiceImpl.java

```diff
@@ -0,0 +1,190 @@
+            return CUTaxConstants.TAX_TYPE_1042S;
+        } else {
+            throw new IllegalStateException("Transaction Detail was not configured for 1099 or 1042-S");
+        }
+    }
+
+    private void initializePayeeIdInformation(final TaxPayeeBase payee, final String payeeId) {
+        final String vendorHeaderId = StringUtils.substringBefore(payeeId, KFSConstants.DASH);
+        final String vendorDetailId = StringUtils.substringAfter(payeeId, KFSConstants.DASH);
+        payee.setPayeeId(payeeId);
+        payee.setVendorHeaderId(Integer.valueOf(vendorHeaderId));
+        payee.setVendorDetailId(Integer.valueOf(vendorDetailId));
+    }
+
+    private void initializeVendorData(final TaxPayeeBase payee, final TaxStatisticsHandler statistics,
+            final String taxType) throws SQLException {
+        final VendorQueryResults vendorResults = transactionDetailProcessorDao.getVendor(
+                payee.getVendorHeaderId(), payee.getVendorDetailId());
+        final VendorDetailLite matchingVendor = vendorResults.getVendor();
+        final VendorDetailLite vendorToProcess;
+
+        if (ObjectUtils.isNull(matchingVendor)) {
+            statistics.increment(TaxStatType.NUM_NO_VENDOR);
+            payee.setVendorName(createMessage(CUTaxKeyConstants.MESSAGE_TAX_OUTPUT_VENDOR_NOT_FOUND,
+                    payee.getVendorHeaderId(), payee.getVendorDetailId()));
+            vendorToProcess = null;
+        } else if (matchingVendor.isVendorParentIndicator()) {
+            vendorToProcess = matchingVendor;
+        } else if (ObjectUtils.isNull(vendorResults.getParentVendor())) {
+            initializeAlternateVendorNameIfSoleProprietor(payee, matchingVendor, taxType);
+            statistics.increment(TaxStatType.NUM_NO_PARENT_VENDOR);
+            payee.setVendorName(createMessage(
+                    CUTaxKeyConstants.MESSAGE_TAX_OUTPUT_VENDOR_PARENT_NOT_FOUND,
+                    payee.getVendorHeaderId(), payee.getVendorDetailId()));
+            vendorToProcess = null;
+        } else {
+            initializeAlternateVendorNameIfSoleProprietor(payee, matchingVendor, taxType);
+            vendorToProcess = vendorResults.getParentVendor();
+        }
+
+        if (ObjectUtils.isNotNull(vendorToProcess)) {
+            payee.setVendorTypeCode(vendorToProcess.getVendorTypeCode());
+            payee.setVendorOwnershipCode(vendorToProcess.getVendorOwnershipCode());
+            payee.setVendorGIIN(vendorToProcess.getVendorGIIN());
+            payee.setVendorChapter4StatusCode(vendorToProcess.getVendorChapter4StatusCode());
+            initializeVendorName(payee, vendorToProcess, statistics);
+            initializeVendorAddressData(payee, vendorToProcess, statistics);
+        } else if (ObjectUtils.isNotNull(matchingVendor)) {
+            initializeVendorAddressData(payee, matchingVendor, statistics);
+        }
+    }
+
+    /*
+     * TODO: We need to revisit the parent-vendor-name setup and determine whether
```

Comment:
Is there. a Jira for this?

Reply:
I don't think a follow-up JIRA has been created for this yet; I'll look into creating one so that I can update or remove this comment.

Reply:
I've created KFSPTS-34474 to follow up on this field name terminology further.